src/statistics.jl

```diff
@@ -1425,21 +1427,21 @@ function apply_statistic(stat::FunctionStatistic,
     # color was bound explicitly
     if aes.color != nothing
         func_color = aes.color
-        aes.color = Array{eltype(aes.color)}(length(aes.y) * stat.num_samples)
-        groups = DataArray(Int, length(aes.y) * stat.num_samples)
+        aes.color = Array{eltype(aes.color)}(undef, length(aes.y) * stat.num_samples)
+        groups = Array{Union{Missing,Int}}(undef, length(aes.y) * stat.num_samples)
```

Comment:
This function looks to me as being quite old, and I don't see why groups needs to contain missing values.